dace/transformation/interstate/loop_lifting.py

```diff
@@ -0,0 +1,99 @@
+
+        loop_info = self.loop_information()
+
+        body = self.loop_body()
+        meta = self.loop_meta_states()
+        full_body = set(body)
+        full_body.update(meta)
+        cond_edge = self.loop_condition_edge()
+        incr_edge = self.loop_increment_edge()
+        inverted = self.expr_index in (2, 3, 5, 6, 7)
+        init_edge = self.loop_init_edge()
+        exit_edge = self.loop_exit_edge()
+
+        label = 'loop_' + first_state.label
+        if loop_info is None:
+            itvar = None
+            init_expr = None
+            incr_expr = None
+        else:
+            incr_expr = f'{loop_info[0]} = {incr_edge.data.assignments[loop_info[0]]}'
+            init_expr = f'{loop_info[0]} = {init_edge.data.assignments[loop_info[0]]}'
+            itvar = loop_info[0]
+
+        left_over_assignments = {}
+        for k in init_edge.data.assignments.keys():
+            if k != itvar:
+                left_over_assignments[k] = init_edge.data.assignments[k]
+        left_over_incr_assignments = {}
+        if incr_edge is not None:
+            for k in incr_edge.data.assignments.keys():
+                if k != itvar:
+                    left_over_incr_assignments[k] = incr_edge.data.assignments[k]
+
+        if inverted and incr_edge is cond_edge:
+            update_before_condition = False
+        else:
+            update_before_condition = True
+
+        loop = LoopRegion(label, condition_expr=cond_edge.data.condition, loop_var=itvar, initialize_expr=init_expr,
+                          update_expr=incr_expr, inverted=inverted, sdfg=sdfg,
+                          update_before_condition=update_before_condition)
+
+        graph.add_node(loop)
+        graph.add_edge(init_edge.src, loop,
+                       InterstateEdge(condition=init_edge.data.condition, assignments=left_over_assignments))
+        graph.add_edge(loop, after, InterstateEdge(assignments=exit_edge.data.assignments))
+
+        loop.add_node(first_state, is_start_block=True)
+        added = set()
+        for e in graph.all_edges(*full_body):
+            if e.src in full_body and e.dst in full_body:
+                if not e in added:
+                    added.add(e)
+                    if e is incr_edge:
+                        if left_over_incr_assignments != {}:
+                            dst = loop.add_state(label + '_tail') if not inverted else e.dst
+                            loop.add_edge(e.src, dst, InterstateEdge(assignments=left_over_incr_assignments))
+                    elif e is cond_edge:
+                        if not inverted:
+                            e.data.condition = properties.CodeBlock('1')
+                            loop.add_edge(e.src, e.dst, e.data)
+                    else:
+                        loop.add_edge(e.src, e.dst, e.data)
+
+        # Remove old loop.
+        for n in full_body:
+            graph.remove_node(n)
+
+        sdfg.recheck_using_experimental_blocks()
```

Comment:
shouldn't this just call `sdfg.using_experimental_blocks()` instead of traversing the entire graph (which can be large?)

Reply:
Also, I look forward to you removing this variable/method in the next PR :)